### PR TITLE
Remove "key" and "value" keys from returned json

### PR DESCRIPTION
**Pull request by patilsnr**
`TwinProperties.ToJson()` will now return something like this:
```json
{
  "properties" : {
      "desired" : {
           "myProp" : "myValue",
           "$version" : 3
      },
     "reported" : {
            "myProp" : "myValue",
           "$version" : 4
     }
   }
}
```

### CODE REVIEW

**Comment by varunpuranik** [comment]
What issue does this address, and why is the new way better than the old one?

**Comment by rido-min** [comment]
Why $version is not a field? where is the $version for reported?

The Json should be something like:

```json
{
  "properties" : {
      "desired" : {
           "myProp" : "myValue",
           "$version" : 3
      },
     "reported" : {
            "myProp" : "myValue",
           "$version" : 4
     }
   }
}
```

**Comment by patilsnr** [comment]
> What issue does this address, and why is the new way better than the old one?

My understanding is that this is consistent with what the V1 API returns. Without this fix an extra level of serialization exists where "key" and "value" are literally written into the json object

File: iothub/device/src/Twin/DesiredProperties.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public class DesiredProperties : PropertyCollection
         /// This class can be inherited from and set by unit tests for mocking purposes.
         /// </remarks>
         protected internal DesiredProperties(Dictionary<string, object> desiredProperties)
-            : base(desiredProperties, true)
+            : base(desiredProperties, false)
```
Comment (patilsnr) [inline iothub/device/src/Twin/DesiredProperties.cs:26]: If this is a mocking ctor, false makes more sense here to just add the properties to the internal dictionary

**Comment by varunpuranik** [comment]
> > What issue does this address, and why is the new way better than the old one?
> 
> My understanding is that this is consistent with what the V1 API returns. Without this fix an extra level of serialization exists where "key" and "value" are literally written into the json object

What is returned by IoT Hub?

**Comment by patilsnr** [comment]
> > > What issue does this address, and why is the new way better than the old one?
> > 
> > 
> > My understanding is that this is consistent with what the V1 API returns. Without this fix an extra level of serialization exists where "key" and "value" are literally written into the json object
> 
> What is returned by IoT Hub?

Running a quick test with a device in my own Hub, I get: 

`{"properties":{"desired":"{\"$version\":\"1\"}","reported":"{\"$version\":\"1\"}"}}`

which is consistent with the desired format

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**